core/common/boot/src/main/java/org/eclipse/edc/boot/system/SyntheticExtension.java

```diff
@@ -0,0 +1,26 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.boot.system;
+
+import org.eclipse.edc.runtime.metamodel.annotation.Provides;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.telemetry.Telemetry;
+import org.eclipse.edc.spi.types.TypeManager;
+
+import java.time.Clock;
+
+@Provides({ Monitor.class, TypeManager.class, Telemetry.class, Clock.class })
+class SyntheticExtension implements org.eclipse.edc.spi.system.ServiceExtension {
```

Comment:
What's the concept behind the `Synthetic` term? 
Could the name be something that explains its use? Like `BaseServicesExtension` or, I don't know, `FoundationalServicesExtension`

Reply:
The extension providing these services is not _really_ providing them, it merely declares the outgoing dependency (`@Provides`), thus acting as a ... well... synthetic provider for them :)
Other than that, it was the first and (to me) most intuitive term, but I have absolutely nothing agains renaming it.

Reply:
Ok, `Dummy` could be a good term to be used as it does not really provide anything but it has to be there for "appearence". What about `BaseServicesDummyExtension`? So we could refer to those services as "base", they are provided out of the box with the `boot` module.

Reply:
Could we name it _just_ `BaseServicesExtension`? Not too keen on having the term `Dummy` in production code...

Reply:
sure, sounds good to me